File: backend/aqua-hackathon/src/main/java/com/k2o/aquahackathon/entity/WaterRequest.java
package com.k2o.aquahackathon.entity;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Data;

import javax.persistence.*;
import java.util.Date;

@Entity
@Data
public class WaterRequest {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    Integer id;

    @ManyToOne
    @JoinColumn(name="user_id")
    User initiator;

    @JsonFormat(pattern="yyyy-MM-dd HH:mm")
    Date createdDate = new Date();

    @JsonFormat(pattern="yyyy-MM-dd HH:mm")
    Date availTill = null;

    Float latitude = 0.0f;

    Float longitude = 0.0f;

    Integer requestType;

    String additionalInfo;


}

File: mobile/test/Model/WaterRequest.swift
//
//  WaterRequest.swift
//  test
//
//  Created by Javid Abbasov on 05.07.2020.
//  Copyright © 2020 Javid Abbasov. All rights reserved.
//

import Foundation

struct WaterRequest: Codable {
    var id: Int?
    var initiator: User?
    
    var createdDate: String?
    var availTill: String?
    var latitude: Double?
    var longitude: Double?
    
    var requestType: Int?
    var additionalInfo: String?
}
